### PR TITLE
Apply changes from upstream delphes: renaming DelphesHepMCReader -> DelphesHepMCReader2

### PR DESCRIPTION
**Pull request by vvolkl**
Normally, I'd try to keep backward compatibility, but due to all the breaking changes in the trackcovariance code, that won't be possible anyway.

BEGINRELEASENOTES
- Apply changes from upstream delphes: renaming DelphesHepMCReader -> DelphesHepMCReader2

ENDRELEASENOTES

### CODE REVIEW

**Comment by vvolkl** [comment]
However, since there is the issue of https://github.com/delphes/delphes/pull/97 in the delphes master, maybe we wait and I switch delphes to 3.4.3pre11 in the nightlies?

**Comment by tmadlener** [comment]
This looks good to me. For full compatibility with delphes we would probably also have to rename our `DelphesHepMC_EDM4HEP` to `DelphesHepMC2_EDM4HEP`, but I think that is too much.

3.4.3pre11 should still work, so we could use that for the nightlies. In the end, I think we have to follow delphes here probably, but I would wait until that has stabilized again, there was quite a bit of development there in the past few weeks. In any case for a k4SimDelphes release we will need a stable delphes release to depend on.

**Review by tmadlener — APPROVED** [review]
Now that we have delphes 3.5.0 this also looks good to me.

File: standalone/src/DelphesHepMCReader.h [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #include "classes/DelphesClasses.h"
 #include "classes/DelphesFactory.h"
 #include "classes/DelphesStream.h"
-#include "classes/DelphesHepMCReader.h"
+#include "classes/DelphesHepMCReader2.h"
```
Comment (tmadlener) [inline standalone/src/DelphesHepMCReader.h:14]: The `2` seems to be at the wrong place here, it should be the following:
```suggestion
#include "classes/DelphesHepMC2Reader.h"
```

File: standalone/src/DelphesHepMCReader.h [hunk 2]
```diff
@@ -37,7 +37,7 @@ class DelphesHepMCInputReader: public DelphesInputReader {
     int i = 4;
 
 
-    m_reader = new DelphesHepMCReader;
+    m_reader = new DelphesHepMCReader2;
```
Comment (tmadlener) [inline standalone/src/DelphesHepMCReader.h:40]:
```suggestion
    m_reader = new DelphesHepMC2Reader;
```

File: standalone/src/DelphesHepMCReader.h [hunk 3]
```diff
@@ -136,7 +136,7 @@ class DelphesHepMCInputReader: public DelphesInputReader {
   FILE *m_inputFile = 0;
   TStopwatch m_readStopWatch, m_procStopWatch;
   ExRootTreeBranch *m_branchEvent = 0, *m_branchWeight = 0;
-  DelphesHepMCReader *m_reader = 0;
+  DelphesHepMCReader2 *m_reader = 0;
```
Comment (tmadlener) [inline standalone/src/DelphesHepMCReader.h:139]:
```suggestion
  DelphesHepMC2Reader *m_reader = 0;
```

**Review by tmadlener — CHANGES_REQUESTED** [review]

**Pull request merged**